packages/venia-concept/src/index.js

```diff
@@ -45,7 +49,28 @@ const apolloLink = ApolloLink.from([
 ReactDOM.render(
     <Adapter apiBase={apiBase} apollo={{ link: apolloLink }} store={store}>
         <AppContextProvider>
+            {/* Some extensions require a target */}
+            <Extension targetId="main-ep-before-children">
+                <div style={{ textAlign: 'center' }}>
+                    I'm not even an extension component, I'm just some DOM!
+                </div>
+            </Extension>
+            <Extension targetId="main-ep-before-children">
+                <div style={{ textAlign: 'center' }}>
+                    I'm an example of an extension trying to overload an
+                    extension point!
+                </div>
+            </Extension>
+            <Extension targetId="main-ep-after-children">
+                <ProductRecommendations />
+            </Extension>
+            {/* Other extensions don't need target (assumes #root at least) */}
+            <Extension>
+                <WelcomeToast />
+            </Extension>
+            {/* App/Portals must be rendered _after_ extensions. */}
```

Comment:
Out of curiosity, why?

And I'm guessing none of the extensions above could have `targetId='root'`?

This begs the question of whether or not extensions will depend on the order of components loading. If so, we'll likely want some code that identifies the cases that won't work and warn about them in development mode. Otherwise we're signing everybody up for lots of debugging nightmares.